### PR TITLE
Rework of README.rst

### PR DESCRIPTION
**Pull request by miclark**
Looking to make the project more accessible to users of insights-core.
Added an addition "walk through" notebook example.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,22 +1,230 @@
-=============
-Insights Core
-=============
+============= Insights Core =============
```
Comment (csams) [inline README.rst]: Heading isn't rendering correctly.

File: README.rst [hunk 2]
```diff
@@ -1,22 +1,230 @@
...
+range of NVRs making the script quite a bit more complicated.
+Especially if we wanted to be able to apply it to other packages in the
+future and the parsing of the NVR and version comparison would need to
+be generalized.
+
+So, how would we code this using `insights-core`?  The following script
+will identify whether the current system has a version of `bash`
+installed within the range.
+
+::
```
Comment (csams) [inline README.rst]: Wonder if we could get this to render with syntax highlighting.

File: README.rst [hunk 3]
```diff
@@ -1,22 +1,230 @@
...
+--------------------------------------
+
+Certainly, for a simple check, just grepping a file is simpler.
+However, there are gains as the complexity and automation requirements
+come into play.  In this particular case, the complexity is hidden in two
+places.
+
+First, complexity around parsing the `rpm -qa` command, and presenting
+it in a usable data model is handled by the `InstalledRpms` class.  It
+is just one(link) of many parsers(link) and combiners(link) that one can use.   In
```
Comment (csams) [inline README.rst]: Missing URL's?

File: README.rst [hunk 4]
```diff
@@ -1,22 +1,230 @@
...
+addition, new custom parsers can be easily created. (See the
+`stand_alone.py` script for a full example defining a full set of
```
Comment (csams) [inline README.rst]: Maybe link to the script in github

File: docs/notebooks/Diagnostic Walkthrough.ipynb [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+   "source": [
+    "from insights import rule, make_response, run\n",
+    "from insights.parsers.installed_rpms import InstalledRpms"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The first import line has the most common components used when creating `insights-core`.  The `rule` decorator marks a function that encodes logic to be applied by the framework and the required or optional components it needs to function.  `@rule` decorated fucntions use `make_response` to pass back results.  The `run` method executes the system, simplifying usage of `insights-core` for small, standalone scripts and from the python interpreter.\n",
```
Comment (csams) [inline docs/notebooks/Diagnostic Walkthrough.ipynb]: Misspelled word: fucntions

File: README.rst (hunk 3 above)
Comment (miclark) [reply README.rst]: Dangit.  Yes, those were reminders that didn't remind me.

File: docs/notebook_index.rst [hunk 6]
```diff
@@ -10,5 +10,6 @@ directory.
 
     notebooks/Datasource Registry.ipynb
     notebooks/Development Examples.ipynb
+    notebooks/Diagnostic Walkthrough.ipynb
```
Comment (bfahr) [inline docs/notebook_index.rst]: @miclark your headers in your notebook become the outline in the docs, so right now the doc title shown in the index is "Checking for Installed rpms".  That may be what you want, but its different from the name of the file, so I'm just checking.

File: README.rst [hunk 7]
```diff
@@ -2,45 +2,267 @@
...
+
+Python 2
+++++++++
+
+To get the project setup for Python 2, use the following commands
+
+.. code-block:: bash
+
+    mkdir .python2
+    virtualenv .  # Make sure you're using the python2 runtime
```
Comment (bfahr) [inline README.rst]: I believe this should be `virtualenv .python2`.

File: README.rst [hunk 8]
```diff
@@ -2,45 +2,267 @@
...
+
+Python 3
+++++++++
+
+To setup the project for Python 3, use the following commands
+
+.. code-block:: bash
+
+    mkdir .python3
+    python3 venv -m .python3
```
Comment (bfahr) [inline README.rst]: This should be `python3 -m venv .python3`.

File: docs/notebook_index.rst (hunk 6 above)
Comment (miclark) [reply docs/notebook_index.rst]: Changed to "Diagnostic Walkthrough" to match the file name.

**Pull request merged**